lib/oanda_api/client/client.rb

```diff
@@ -158,9 +170,11 @@ def self.last_throttled_at
     # @param [Time] time The time that the throttle was requested.
     #
     # @return [void]
-    def self._throttle(time)
-      @throttle_mutex.synchronize { @throttled_at = time }
-      sleep OandaAPI.configuration.min_request_interval
+    def self._throttle(delta, time)
+      @throttle_mutex.synchronize do
+        @throttled_at = time
+        sleep OandaAPI.configuration.min_request_interval - delta
```

Comment:
I said throttling only works if you sleep inside the mutex, which is true. The way the logic is now, the request limit is still violated. It would only be a deadlock if some other part of the application slept in another lock and waited until the first lock was released while a different thread was doing the reverse. This is not the case.